src/libdredd/src/mutation_replace_expr.cc

```diff
@@ -538,18 +538,18 @@ void MutationReplaceExpr::ReplaceExprWithFunctionCall(
     suffix.append("; }");
   }
 
-  if (!ast_context.getLangOpts().CPlusPlus) {
-    if (expr_->isLValue() && input_type.ends_with('*')) {
-      prefix.append("&(");
-      suffix.append(")");
-    } else if (const auto* binary_expr =
-                   llvm::dyn_cast<clang::BinaryOperator>(expr_)) {
-      // The comma operator requires special care in C, to avoid it appearing to
-      // provide multiple parameters for an enclosing function call.
-      if (binary_expr->isCommaOp()) {
-        prefix.append("(");
-        suffix.append(")");
-      }
+  if (!ast_context.getLangOpts().CPlusPlus && expr_->isLValue() &&
+      input_type.ends_with('*')) {
+    prefix.append("&(");
+    suffix = ")" + suffix;
+  }
+
+  if (const auto* binary_expr = llvm::dyn_cast<clang::BinaryOperator>(expr_)) {
+    // The comma operator requires special care in C, to avoid it appearing to
+    // provide multiple parameters for an enclosing function call.
+    if (binary_expr->isCommaOp()) {
+      prefix.append("(");
+      suffix = ")" + suffix;
```

Comment:
I'm sure you're right with this change. It's not all that obvious to me, but I'm not "in the zone" with this at present. I see you have lots of tests, which is great.

Reply:
I seem to remember that it's the case that something being a C LValue is not a mutually exclusive check from wrapping a comma operator with brackets. Hence the issue occurring when initializing variables using a comma operator.